tasks/task_manager.cc

```diff
@@ -183,7 +184,7 @@ void task_manager::task::change_state(task_state state) noexcept {
 }
 
 void task_manager::task::add_child(foreign_task_ptr&& child) {
-    _impl->_children.push_back(std::move(child));
+    _impl->_children.emplace(child->id(), std::move(child));
```

Comment:
Is it possible that the child id would already exist?
emplace won't override it...